java/monkey/src/main/java/root/ast/Identifier.java

```diff
@@ -0,0 +1,18 @@
+package root.ast;
+
+import root.Token;
+
+public class Identifier extends Expression {
```

Comment:
This is incorrect to the Book.
We can see on Page 45 that Identifier != Expression, as they branch on the `LetStatement` in the AST.

It should _not_ extend Expression.

Reply:
I do believe that an Identifier should be an Expression, as it is capable of generating a value, as opposed to an let statement

<img width="665" alt="image" src="https://github.com/ThePrimeagen/ts-rust-zig-deez/assets/39140806/6ac96323-42f2-44af-b26f-3f7b4fbbde4d">

We can see in the book that while `LetStatement` implements the `statementNode` function,`Identifier` implements the `expressionNode` function.

Reply:
In that case swap the name to  `IdentiferExpression` to more closely align with more typical Java Naming Conventions.